pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -113,18 +124,12 @@ def restore(self, checkpoint_path: str, on_gpu: bool):
         # restore the state_dict on the model
         model.load_state_dict(checkpoint['state_dict'])
 
-        if on_gpu:
-            model.cuda(self.trainer.root_gpu)
-
         # restore amp scaling
         if self.trainer.amp_backend == AMPType.NATIVE and 'native_amp_scaling_state' in checkpoint:
             self.trainer.scaler.load_state_dict(checkpoint['native_amp_scaling_state'])
         elif self.trainer.amp_backend == AMPType.APEX and 'amp_scaling_state' in checkpoint:
             amp.load_state_dict(checkpoint['amp_scaling_state'])
```

Comment:
should this be in `restore_training_state`?

Reply:
Before my PR, amp state **was already placed outside** the `restore_training_state`.  
There could be some implicit reason (although I cannot found any reason).  
For conservative code modification, I placed it in `restore_model_states`.  
This can be moved in `restore_training_state` in future separated PR with detailed side-effect survey.

Reply:
I think it has more to do with training and not model itself, so I though it should be there.

Reply:
I checked side-effect deeper.  
`restore_training_state` is used only in test, and this test do not touch amp related things.  
amp restore is moved proper place as you mentioned, thanks for great suggestion👍